cmd/create/cluster/cmd.go

```diff
@@ -418,40 +399,98 @@ func run(cmd *cobra.Command, _ []string) {
 
 	// AWS ARN Role
 	roleARN := args.roleARN
-	if roleARN != "" && interactive.Enabled() {
+
+	if !interactive.Enabled() && creator.IsSTS && roleARN == "" {
+		err = interactive.PrintHelp(interactive.Help{
+			Message: "Since your AWS credentials are returning an STS ARN you can only " +
+				"create STS clusters. Otherwise, switch to IAM credentials.",
+		})
+		if err != nil {
+			reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		interactive.Enable()
+		reporter.Infof("Enabling interactive mode")
+	}
```

Comment:
@jharrington22 - just added this to ensure that STS credentials without a Role ARN have to enter it interactively.